Feature/uprobe for https

yxshen20 opened the pull request:
This commit includes the implementation of uprobes for HTTPS observability

Commit:
initial commit for uprobe_https

datawine commented:
This PR needs to be resubmitted to address issues with commit formatting, licensing, file directory structure, and the README, among other aspects.

Commit:
[Style] added license and comments to adhere to the apache 2.0 license

Commit:
[Style] added license (#12)

thu-gyt commented:
The same problem
```
sudo go run ./sniffer/main.go ./sniffer/ebpf_src.c --pid $(pgrep -f "python3 ./server/https_server.py 43421")
# github.com/iovisor/gobpf/bcc
/root/go/pkg/mod/github.com/iovisor/gobpf@v0.2.0/bcc/module.go:32:10: fatal error: bcc/bcc_common.h: No such file or directory
   32 | #include <bcc/bcc_common.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

Commit:
[Dependencies] Added dependencies for ebpf http observability

Commit:
[Dependencies] Added dependencies for ebpf http observability (#12)

Commit:
[Documentation] Updated README file to a more detailed version  (#12)

yxshen20 commented:
Addressed the following issues:
1. Added LICENSE and comments to the source files, adhering to the policies of Apache 2.0
2. Added scripts for installing depencencies
3. Updated to a more detailed README file

thu-gyt committed:
Merge branch 'feature/http-observability' into feature/uprobe-for-https
